lib/nostrum/api.ex

```diff
@@ -4462,31 +3774,36 @@ defmodule Nostrum.Api do
   end
 
   # If `:embed` is present, prepend to `:embeds` for compatibility
-  defp combine_embeds(%{embed: embed} = args),
+  def combine_embeds(%{embed: embed} = args),
     do: Map.delete(args, :embed) |> Map.put(:embeds, [embed | args[:embeds] || []])
 
-  defp combine_embeds(%{data: data} = args), do: %{args | data: combine_embeds(data)}
-  defp combine_embeds(%{message: data} = args), do: %{args | message: combine_embeds(data)}
-  defp combine_embeds(args), do: args
+  def combine_embeds(%{data: data} = args), do: %{args | data: combine_embeds(data)}
+  def combine_embeds(%{message: data} = args), do: %{args | message: combine_embeds(data)}
+  def combine_embeds(args), do: args
 
   # If `:file` is present, prepend to `:files` for compatibility
-  defp combine_files(%{file: file} = args),
+  def combine_files(%{file: file} = args),
     do: Map.delete(args, :file) |> Map.put(:files, [file | args[:files] || []])
 
-  defp combine_files(%{data: data} = args), do: %{args | data: combine_files(data)}
-  defp combine_files(%{message: data} = args), do: %{args | message: combine_files(data)}
-  defp combine_files(args), do: args
+  def combine_files(%{data: data} = args), do: %{args | data: combine_files(data)}
+  def combine_files(%{message: data} = args), do: %{args | message: combine_files(data)}
+  def combine_files(args), do: args
 
-  defp pop_files(%{data: data} = args),
+  def pop_files(%{data: data} = args),
     do: {data.files, %{args | data: Map.delete(data, :files)}}
 
-  defp pop_files(%{message: data} = args),
+  def pop_files(%{message: data} = args),
     do: {data.files, %{args | message: Map.delete(data, :files)}}
 
-  defp pop_files(args), do: Map.pop!(args, :files)
+  def pop_files(args), do: Map.pop!(args, :files)
 
   @doc false
   def bangify(to_bang) do
+    Logger.warning("""
+    Using the bangified version of the Api call is deprecated and will be removed in the next major release.
+    Please use the non-bangified version and handle the error case yourself.
+    """)
```

Comment:
What if instead of putting the `Logger.warning` call here, you add `@deprecated` tags to most/all the functions in this module? Idea being that with the following major release we remove most/all of this module and point users to the broken out functions.

That way we don't pollute user's logs needlessly and they just get a warning once at compile time.